ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/wizards/plugin/NewProjectCreationOperation.java

```diff
@@ -554,6 +555,15 @@ private IPluginReference[] getDependencies() {
 			IPluginReference[] refs = fGenerator.getDependencies();
 			Collections.addAll(result, refs);
 		}
+		// Else if a plug-in project is being created...
+		else if (fData instanceof IPluginFieldData) {
+			// And the plug-in project makes contributions to the UI...
+			if (((IPluginFieldData) fData).isUIPlugin()) {
+				// Add SWT and JFace as dependencies
+				result.add(new PluginReference("org.eclipse.swt")); //$NON-NLS-1$
+				result.add(new PluginReference("org.eclipse.jface")); //$NON-NLS-1$
+			}
+		}
```

Comment:
You can use Pattern matching for instanceof here and then combine the two if statements nicely into one.
Furthermore the comments can be reduced here. It's not necessary to write down every action in a corresponding comment. Comments should only be used to clarify the logic where it is not obvious from the code (ideally the code is readable and understandable in itself).

And last but not least the logic should be enhanced to to not add the dependencies if a user has selected a template.
If a template is selected that template already defines the necessary dependencies and nothing should be change here.
Just as described in https://github.com/eclipse-pde/eclipse.pde/issues/1306#issuecomment-2176489533.

```suggestion
		} else if (fContentWizard == null && fData instanceof IPluginFieldData pluginData && pluginData.isUIPlugin()) {
			// a plug-in project is being created and makes contributions to UI
			result.add(new PluginReference("org.eclipse.swt")); //$NON-NLS-1$
			result.add(new PluginReference("org.eclipse.jface")); //$NON-NLS-1$
		}
```

Reply:
Thanks for the suggestions! I added the additional check for the use of templates and used pattern matching as you suggested. I have appended the commit and pushed it to this branch.